[32m2023-12-18 17:04:50 +0000[0m - dagster - [34mDEBUG[0m - __ASSET_JOB - 538f7ceb-a3ca-4793-ba6b-88fc2f538543 - 12732 - LOGS_CAPTURED - Started capturing logs in process (pid: 12732).
[32m2023-12-18 17:04:50 +0000[0m - dagster - [34mDEBUG[0m - __ASSET_JOB - 538f7ceb-a3ca-4793-ba6b-88fc2f538543 - 12732 - SalaryDistribPerDept - STEP_START - Started execution of step "SalaryDistribPerDept".
[32m2023-12-18 17:04:50 +0000[0m - dagster - [34mDEBUG[0m - __ASSET_JOB - 538f7ceb-a3ca-4793-ba6b-88fc2f538543 - SalaryDistribPerDept - Loading file from: /Users/nirmalgeorge/dap_dagster_project/DAP-project/govt_employee_analysis/tmpjy9nrer1/storage/LoadingCsv using PickledObjectFilesystemIOManager...
[32m2023-12-18 17:04:50 +0000[0m - dagster - [34mDEBUG[0m - __ASSET_JOB - 538f7ceb-a3ca-4793-ba6b-88fc2f538543 - 12732 - SalaryDistribPerDept - ASSET_OBSERVATION - DagsterEventType.ASSET_OBSERVATION for step SalaryDistribPerDept
[32m2023-12-18 17:04:50 +0000[0m - dagster - [34mDEBUG[0m - __ASSET_JOB - 538f7ceb-a3ca-4793-ba6b-88fc2f538543 - 12732 - SalaryDistribPerDept - LOADED_INPUT - Loaded input "LoadingCsv" using input manager "io_manager"
[32m2023-12-18 17:04:50 +0000[0m - dagster - [34mDEBUG[0m - __ASSET_JOB - 538f7ceb-a3ca-4793-ba6b-88fc2f538543 - 12732 - SalaryDistribPerDept - STEP_INPUT - Got input "LoadingCsv" of type "Any". (Type check passed).
[32m2023-12-18 17:04:50 +0000[0m - dagster - [34mERROR[0m - [31m__ASSET_JOB - 538f7ceb-a3ca-4793-ba6b-88fc2f538543 - 12732 - SalaryDistribPerDept - STEP_FAILURE - Execution of step "SalaryDistribPerDept" failed.

dagster._core.errors.DagsterExecutionStepExecutionError: Error occurred while executing op "SalaryDistribPerDept"::

NameError: name 'context' is not defined

Stack Trace:
  File "/Users/nirmalgeorge/anaconda3/lib/python3.11/site-packages/dagster/_core/execution/plan/utils.py", line 54, in op_execution_error_boundary
    yield
  File "/Users/nirmalgeorge/anaconda3/lib/python3.11/site-packages/dagster/_utils/__init__.py", line 465, in iterate_with_context
    next_output = next(iterator)
                  ^^^^^^^^^^^^^^
  File "/Users/nirmalgeorge/anaconda3/lib/python3.11/site-packages/dagster/_core/execution/plan/compute_generator.py", line 131, in _coerce_op_compute_fn_to_iterator
    result = invoke_compute_fn(
             ^^^^^^^^^^^^^^^^^^
  File "/Users/nirmalgeorge/anaconda3/lib/python3.11/site-packages/dagster/_core/execution/plan/compute_generator.py", line 125, in invoke_compute_fn
    return fn(context, **args_to_pass) if context_arg_provided else fn(**args_to_pass)
                                                                    ^^^^^^^^^^^^^^^^^^
  File "/Users/nirmalgeorge/dap_dagster_project/DAP-project/govt_employee_analysis/govt_employee_analysis/assets.py", line 163, in SalaryDistribPerDept
    context.add_output_metadata(
    ^^^^^^^
[0m
